src/rules/no-test-callback.ts

```diff
@@ -20,24 +25,36 @@ export default {
 
         const [, callback] = node.arguments;
 
-        if (
-          !/^(Arrow)?FunctionExpression$/.test(callback.type) ||
-          callback.params.length !== 1
-        ) {
+        if (!isFunction(callback) || callback.params.length !== 1) {
           return;
         }
 
         const [argument] = callback.params;
+        const { body } = callback;
+
+        if (!body || !('name' in argument)) {
```

Comment:
We want to report either way, don't we? If we only use this in the fixer the checks should be moved into it

Reply:
Yeah, I originally had them there b/c otherwise I have to duplicate the throw, since `body` is used for `firstBodyToken` & `lastBodyToken`.